plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1066 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Frank Dornheim <dornheim@posteo.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: lxd
+    short_description: Returns Ansible inventory from lxd host
+    description:
+        - Get inventory from the lxd.
+        - Uses a YAML configuration file that ends with 'lxd.(yml|yaml)'.
+    version_added: "2.5.0"
+    author: "Frank Dornheim (@conloos)"
+    options:
+        plugin:
+            description: Token that ensures this is a source file for the 'lxd' plugin.
+            required: true
+            choices: [ 'community.general.lxd' ]
+        url:
+            description:
+            - The unix domain socket path or the https URL for the lxd server.
+            - Sockets in filesystem have to start with C('unix:').
+            - Mostly C(unix:/var/lib/lxd/unix.socket) or C(unix:/var/snap/lxd/common/lxd/unix.socket).
+            default: unix:/var/snap/lxd/common/lxd/unix.socket
+            required: false
```

Comment:
```suggestion
```
In general, you should not specify things like this, or `default: none`.

Reply:
At the moment i use that default values to test if a config option is set.
I you don't like it i have to use try and ecept KeyError ...

```
> /home/BODOMOS/frank/Workspace/ansible_lxd_test/inventory_plugins/lxd.py(1047)parse()
-> self.client_key = self.get_option('client_key')
(Pdb) self.get_option('tralala')
*** KeyError: 'Requested entry (plugin_type: inventory plugin: lxd setting: tralala ) was not defined in configuration.'
```

Reply:
I don't think this error is related to `required: false` not being there. As you can see in the code which interprets this: https://github.com/ansible/ansible/blob/devel/lib/ansible/config/manager.py#L530 it does not care whether `required` is actually present.